File: AlgoExpert/Dynamic Programming/Medium/Max Subset Sum No Adjacent.java
import java.util.*;

class Program {
	public static int maxSubsetSumNoAdjacent(int[] array) {
		// Write your code here.
		if (array.length == 0) return 0;
		if (array.length == 1) return array[0];

		int first = array[0];
		int second = Math.max(array[0], array[1]);

		for (int i = 2; i < array.length; i++) {
			int current = Math.max(array[i] + first, second);
			first = second;
			second = current;
		}

		return second;
	}
}

File: Python/AlgoExpert/Dynamic Programming/Medium/maxSubsetSumNoAdjacent.py
def maxSubsetSumNoAdjacent(array):
    # Write your code here.
    if len(array) == 0:
        return 0
    elif len(array) == 1:
        return array[0]

    maximums = [array[0], max(array[0], array[1])]

    for i in range(2, len(array)):
        maximums.append(max(maximums[i - 1], array[i] + maximums[i - 2]))

    return maximums[len(maximums) - 1]
